core/lib/snapshots_applier/src/lib.rs

```diff
@@ -246,6 +261,12 @@ impl SnapshotsApplierTask {
         }
     }
 
+    /// Specifies the L1 batch to recover from. This setting is ignored if recovery is complete, but is checked
+    /// if recovery is in progress (so if a node started recovering from another snapshot, it will error).
```

Comment:
I'd expect such setting to only impact the recovery initialization, the same way the snapshots recovery behaves if the newest snapshot changes